python/addBTV.py

```diff
@@ -216,12 +235,48 @@ def get_DeepCSV_vars():
     )
     return DeepCSVVars
 
-def add_BTV(process, runOnMC=False, onlyAK4=False, onlyAK8=False, keepInputs=True):
+
+def get_DeepJet_outputs():
+    DeepJetOutputVars = cms.PSet(
+        btagDeepFlavB_b=Var("bDiscriminator('pfDeepFlavourJetTags:probb')",
+                            float,
+                            doc="DeepJet b tag probability",
+                            precision=10),
+        btagDeepFlavB_bb=Var("bDiscriminator('pfDeepFlavourJetTags:probbb')",
+                             float,
+                             doc="DeepJet bb tag probability",
+                             precision=10),
+        btagDeepFlavB_lepb=Var("bDiscriminator('pfDeepFlavourJetTags:problepb')",
+                               float,
+                               doc="DeepJet lepb tag probability",
+                               precision=10),
+        btagDeepFlavC=Var("bDiscriminator('pfDeepFlavourJetTags:probc')",
+                            float,
+                            doc="DeepJet c tag probability",
+                            precision=10),
+        btagDeepFlavUDS=Var("bDiscriminator('pfDeepFlavourJetTags:probuds')",
+                            float,
+                            doc="DeepJet uds tag probability",
+                            precision=10),
+        btagDeepFlavG=Var("bDiscriminator('pfDeepFlavourJetTags:probg')",
+                          float,
+                          doc="DeepJet gluon tag probability",
+                          precision=10),
+        # anstein: could add discriminators here (or just leave it as it is, only output nodes with individual probabilities)
```

Comment:
I would add the discriminators, it's just 3 branches anyway, B, Cvl, CvB

Reply:
Just saw that these discriminators are already included via nano (declared inside `jets_cff.py`):

```
'Jet_btagDeepFlavB',
'Jet_btagDeepFlavC',  # this one only until 106x
'Jet_btagDeepFlavCvB',
'Jet_btagDeepFlavCvL',
'Jet_btagDeepFlavQG',
```

So I can probably keep it as it is? (`btagDeepFlavC` will be duplicated, depending on the era (<= 106x), discriminators are always in, not dependent on the era, individual probabilities I add manually, not dependent on the era)

See
[https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/jets_cff.py#L233-L239](https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/jets_cff.py#L233-L239)
for the discriminators

and
[https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/jets_cff.py#L268](https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/jets_cff.py#L268)
for btagDeepFlavC

Reply:
👍 indeed, in fact if you could check there aren't any duplicate branches in the current output fill overall that would be great

Reply:
```
def duplicates(mykeys):
    seen = set()
    for x in mykeys:
        if x in seen: 
            print(x)
        seen.add(x)
    return

duplicates(keys)
```

used for the configuration `PFnano_customizeMC_add_DeepJet` gives me this output

```
FatJet_btagDDBvLV2
FatJet_btagDDCvBV2
FatJet_btagDDCvLV2
Jet_btagDeepFlavC
FatJet_nBHadrons
FatJet_nCHadrons
SubJet_nBHadrons
SubJet_nCHadrons
```

`Jet_btagDeepFlavC` was expected, as explained above, the others are not related to what I added or modified in this PR I guess.